src/mono/mono/mini/mini-llvm.c

```diff
@@ -9104,11 +9112,18 @@ process_bb (EmitContext *ctx, MonoBasicBlock *bb)
 			case SIMD_OP_LLVM_I16ABS: id = INTRINS_AARCH64_ADV_SIMD_ABS_INT16; break;
 			case SIMD_OP_LLVM_I32ABS: id = INTRINS_AARCH64_ADV_SIMD_ABS_INT32; break;
 			case SIMD_OP_LLVM_I64ABS: id = INTRINS_AARCH64_ADV_SIMD_ABS_INT64; break;
+			case SIMD_OP_ARM64_SHA1H: id = INTRINS_AARCH64_SHA1H; getLowerElement = TRUE; break;
 			default: g_assert_not_reached (); break;
 			}
-
 			LLVMValueRef arg0 = lhs;
-			values [ins->dreg] = call_intrins (ctx, id, &arg0, "");
+			LLVMValueRef result;
+			if (getLowerElement)
+				arg0 = LLVMBuildExtractElement (ctx->builder, arg0, const_int32 (0), "");
+			result = call_intrins (ctx, id, &arg0, "");
+			if (getLowerElement)
+				result = LLVMBuildZExt (ctx->builder, result, LLVMInt64Type (), "");
```

Comment:
Wouldn't you want a <i32, 2> instead of an i64 here?

Reply:
You are right. I've fixed it.